includes/sanitizers/class-amp-base-sanitizer.php

```diff
@@ -257,21 +257,59 @@ public function sanitize_dimension( $value, $dimension ) {
 	/**
 	 * Sets the layout, and possibly the 'height' and 'width' attributes.
 	 *
-	 * @param string[] $attributes {
+	 * @param array $attributes {
 	 *      Attributes.
 	 *
-	 *      @type int $height
-	 *      @type int $width
-	 *      @type string $sizes
-	 *      @type string $class
-	 *      @type string $layout
+	 *      @type int|string $height
+	 *      @type int|string $width
+	 *      @type string     $sizes
+	 *      @type string     $class
+	 *      @type string     $layout
 	 * }
 	 * @return array Attributes.
 	 */
 	public function set_layout( $attributes ) {
 		if ( isset( $attributes['layout'] ) && ( 'fill' === $attributes['layout'] || 'flex-item' !== $attributes['layout'] ) ) {
 			return $attributes;
 		}
+
+		// Special-case handling for inline style that should be transformed into layout=fill.
+		if ( ! empty( $attributes['style'] ) ) {
+			$styles = $this->parse_style_string( $attributes['style'] );
+
+			// top, left, bottom, right are used.
+			if ( isset( $styles['position'], $styles['top'], $styles['left'], $styles['bottom'], $styles['right'] )
+				&& 'absolute' === $styles['position']
+				&& '0' === (string) $styles['top']
+				&& '0' === (string) $styles['left']
+				&& '0' === (string) $styles['bottom']
+				&& '0' === (string) $styles['right']
```

Comment:
Perhaps it would be good to `intval()` for each of these instead? Sometimes people do `0px` even though they shouldn't. And `0 === intval( '0px' )`

Reply:
Yes, makes sense. I'll use a cast `(int)` instead, though. `intval()` is more of a remnant from PHP 4 when actual type-casting was not yet part of the language.

Reply:
Fixed in https://github.com/ampproject/amp-wp/pull/3209/commits/94156e5c02bf5a2d6764f0ef3853348e62747844